[#2550][DOC] Add instruction for EKS deployment (#3939)
pachyderm/pachyderm: doc/deployment/amazon_web_services/index.rst

.. _deploy-aws:

Deploy Pachyderm on Amazon AWS
==============================

Pachyderm can run in a Kubernetes cluster deployed in Amazon®
Web Services (AWS), whether it is an Elastic Container

> Service (EKS) or a Kubernetes cluster deployed by a deployment tool.

Reviewer: ```suggestion
Service (EKS) or a Kubernetes cluster deployed directly on EC2 using a deployment tool.
```

> AWS removes the need to maintain the underlying virtual cloud.

Reviewer: People know what AWS is, no? 

"AWS removes the need to maintain the underlying virtual cloud." Is of course a true statement, but I think is obvious. While I understand I'm not one to talk in this regard, let's err on the side of only writing what's needed and keeping the docs tighter.

This advantage makes AWS a logical choice for organizations that
decide to offload the cloud infrastructure operational burden
to a third-party vendor. Pachyderm seamlessly integrates with
Amazon EKS and runs in the same fashion as on your computer.

You can install Pachyderm on Amazon AWS by using one of the following
options:

> Deploy Pachyderm by using Amazon EKS

Reviewer: ```suggestion
Deploy Pachyderm on Amazon EKS
```

Reviewer: Also, should this be a link to the EKS doc?

 If you already have an Amazon EKS cluster, you can quickly deploy
 Pachyderm on top of it. If you are just starting with Amazon EKS,
 this section guides you through the EKS deployment process.

Deploy Pachyderm on Amazon EC2 by using ``kops``

>  Another way to deploy Kubernetes on AWS, is to use a

Reviewer: ```suggestion
Instead of EKS, you can deploy Kubernetes on AWS EC2 directly using a Kubernetes
 deployment tool such as ``kops`` (INSERT LINK TO KOPS GITHUB OR DOCS?) and then deploy Pachyderm on that Kubernetes cluster. If you deploy a cluster with `kops`, you
 remain responsible for the Kubernetes operations and maintenance.```

 deployment tool. If you cannot use, Amazon EKS, you can deploy
 Kubernetes by using ``kops`` and then deploy Pachyderm on that
 Kubernetes cluster. If you deploy a cluster with `kops`, you
 remain responsible for the Kubernetes operations and maintenance.

> Deploy by executing a deployment script

Reviewer: ```suggestion
Deploy with a quickstart script
```

>  This option helps you to start quickly and test Pachyderm

Reviewer: ```suggestion
This option helps you to start quickly and test Pachyderm
 functionalities in an AWS environment. The deployment script
 installs both Kubernetes and Pachyderm in a single run. This is not meant as a production deployment option, it's an easy way to get started on AWS for users who are unable to use EKS.
```

Honestly, I'm on the fence about whether or not we just remove this completely. Between EKS and Kops, I don't feel like this providing much value anymore. remove?

 functionalities in an EKS environment. The deployment script
 installs both EKS and Pachyderm in a single run.

> Deploy Pachyderm with CloudFront

Reviewer: The fact that the cloudfront deploy requires them to use the aws script is kinda silly and just a carryover from when our deploy script was the most reasonable way to actually deploy k8s on AWS (these days its the least reasonable way :P). 

We'll have to check with eng on this, but we might be able to remove all the deployment parts before line 47 in the cloudfront doc. I don't really know anything about cloudfront, but presumably you can just turn it on for any existing pachyderm/k8s/s3 deployment you already have. You might need to still do the keypair stuff described here, but that also could be outdated.

Author: I've changed the section to exclude the quickstart script and use the Pachyderm deployment instructions for AWS. CloudFront can be enabled in the AWS console but the part with the key pair remains if the user wants to create signed URLs and cookies. I will try to test this as soon as I get access to the key pair.
